### PR TITLE
XW-918 | Update article.js to get new API response; unify UA dimensions set up

### PR DESCRIPTION
**Pull request by vforge**
https://wikia-inc.atlassian.net/browse/XW-918

https://github.com/Wikia/app/pull/9554

### CODE REVIEW

File: front/common/modules/Trackers/UniversalAnalytics.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ class UniversalAnalytics {
 		this.accountPrimary = 'primary';
 		this.accountSpecial = 'special';
 		this.accountAds = 'ads';
+		this.dimensionsUpdated = false;
```
Comment (hakubo) [inline front/common/modules/Trackers/UniversalAnalytics.js]: why do we need this flag?

File: front/main/app/models/article.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import Ember from 'ember';
 import MediaModel from './media';
 import {normalizeToWhitespace} from 'common/utils/string';
+import UniversalAnalytics from 'common/modules/Trackers/UniversalAnalytics';
```
Comment (hakubo) [inline front/main/app/models/article.js]: This is weird that model knows about UA...
maybe we could move it at least to something like article-content component?

File: front/common/modules/Trackers/UniversalAnalytics.js (hunk 1 above)
Comment (vforge) [inline front/common/modules/Trackers/UniversalAnalytics.js]: Timeline:
(1) dimensions are being set up in Ember initializer
(2) trackers are being initialized
(3) **later** article is fetched from api - and one dimension is set there
(4) track for page view *OR* event is being sent to the server

My changes:
A) Before my changes dimensions were set in (2), but right now we're waiting for (4)
B) Before my changes dimensions were set up multiple times, right now it's only each time we've changed `UniversalAnalytics.dimensions`, before track event - that's why we need a flag answering the quesion: "is anything changed and we need to re-set dimensions?"

File: front/main/app/models/article.js (hunk 2 above)
Comment (vforge) [inline front/main/app/models/article.js]: I'm thinking that this solution is simplest one - we're stopped polluting M.prop with useless property and we're setting dimension only and directly when we need it.

Comment (vforge) [inline front/main/app/models/article.js]: And as we know, we'll improve tracking in the future, I think it's enough for time being.

File: front/main/app/instance-initializers/setup-tracking.js [hunk 3]
```diff
@@ -46,7 +46,7 @@ export function initialize() {
 	// TODO: Krux segmenting not implemented in Mercury https://wikia-inc.atlassian.net/browse/HG-456
 	// ga(prefix + 'set', 'dimension16', getKruxSegment());
 	dimensions[17] = Mercury.wiki.vertical;
-	dimensions[19] = M.prop('article.type');
+	// dimension[19] - articleType is being set in models/article.js
```
Comment (rogatty) [inline front/main/app/instance-initializers/setup-tracking.js]: `articleType is being updated in models/article.js on every article change`?

File: front/main/app/models/article.js [hunk 4]
```diff
@@ -206,13 +207,12 @@ ArticleModel.reopenClass({
 				articleProperties.mainPageData = data.mainPageData;
 				articleProperties.isCuratedMainPage = true;
 			}
+
+			if (data.articleType) {
+				UniversalAnalytics.setDimension(19, articleProperties.articleType);
```
Comment (rogatty) [inline front/main/app/models/article.js]: Remember about the cache. Let's have a fallback for `articleProperties.type` with a ticket number to clean it up.

File: front/main/app/models/article.js (hunk 2 above)
Comment (rogatty) [inline front/main/app/models/article.js]: I agree with @hakubo. This should be done in `components/article-wrapper.js#performArticleTransforms()` where the rest of tracking logic is. Also please use this opportunity to rename this function to something more appropriate ;)

No need to use `M.prop`.

File: front/main/app/instance-initializers/setup-tracking.js (hunk 3 above)
Comment (vforge) [inline front/main/app/instance-initializers/setup-tracking.js]: Yes.

**Comment by hakubo** [comment]
LGTM

**Comment by vforge** [comment]
@nandy-andy can you LGTM this change: https://github.com/Wikia/mercury/commit/5a5fe03a5c04d7f709f866bb645df4cfaa3a1d69 ?

**Pull request merged**